core/arch/arm/tee/entry_std.c

```diff
@@ -221,6 +222,12 @@ static void entry_invoke_command(struct thread_smc_args *smc_args,
 	struct tee_dispatch_invoke_command_out out;
 	struct optee_msg_param *params = OPTEE_MSG_GET_PARAMS(arg);
 
+#ifdef CFG_TEE_BENCHMARK
+	if (params[TEE_BENCH_DEF_PARAM].u.tmem.buf_ptr)
+		tee_add_timestamp((void *)(uintptr_t)
+			params[TEE_BENCH_DEF_PARAM].u.tmem.buf_ptr,
+			TEE_BENCH_CORE);
+#endif
```

Comment:
i think each layer show provides its own `optee_add_timestamp()` with the stamp-ID that show which binary put the stamp.

Thinking about it, linux driver and optee core do change the `params` content (va/pa conversion). Hence, these at least need to know if `params` stores a PA or a VA.

I would rather name it `optee_benchmark_add_stamp()` as it is "optee benchmark" support.